src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,122 @@
+package duke;
+
+import java.util.Scanner;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.MarkCommand;
+import duke.command.ShowListCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Parser to parse the input string and determine its command.
+ * CS2103T iP
+ * AY22/23 Semester 1
+ *
+ * @author Perry Wong
+ */
+public class Parser {
+
+    //Exception Messages
+    private static final String NO_DESC_TODO = "☹ OOPS!!! Description of the todo can't be empty.";
+    private static final String NO_DESC_DEADLINE = "☹ OOPS!!! Description of the deadline can't be empty.";
+    private static final String NO_DESC_EVENT = "☹ OOPS!!! Description of the event can't be empty.";
+    private static final String NO_DATE_DEADLINE = "☹ OOPS!!! Date/time of the deadline can't be empty.";
+    private static final String NO_DATE_EVENT = "☹ OOPS!!! Date/time of the event can't be empty.";
+    private static final String NO_INDEX = "☹ OOPS!!! Index of the task to be marked/unmarked/deleted can't be empty.";
+    private static final String NO_TARGET = "☹ OOPS!!! Keyword of task to be found can't be empty.";
+    private static final String UNKNOWN_COMMAND = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+
+    /**
+     * Processes the input string and returns its corresponding Command.
+     *
+     * @param command Input string that denotes a command for the chatbot.
+     * @return Command that the string is referring to.
+     * @throws DukeException If the input string is invalid and does not represent a valid command.
+     */
+    public static Command parse(String command) throws DukeException {
+        Scanner sc = new Scanner(command);
+
+        if (sc.hasNext("mark")) {
```

Comment:
Would a `switch` be better here?

Reply:
Thanks for the suggestion! I have changed it to `switch` and it looks neater now.